saleor/graphql/checkout/mutations.py

```diff
@@ -145,18 +149,18 @@ def check_lines_quantity(variants, quantities, country, channel_slug):
         raise ValidationError({"quantity": errors})
 
 
-def validate_variants_available_for_purchase(variants, channel_id):
-    not_available_variants = []
-    for variant in variants:
-        product_channel_listing = variant.product.channel_listings.filter(
-            channel_id=channel_id
-        ).first()
-        if not (
-            product_channel_listing
-            and product_channel_listing.is_available_for_purchase()
-        ):
-            not_available_variants.append(variant.pk)
-
+def validate_variants_available_for_purchase(variants_id: set, channel_id: str):
+    today = datetime.date.today()
+    is_available_for_purchase = Q(
+        available_for_purchase__isnull=False,
```

Comment:
I don't think that this condition is required, when you're applying filter for `__lt`
```
In [21]: ProductChannelListing.objects.filter(available_for_purchase__lt=today).count()
Out[21]: 62
Number of queries 1

In [22]: c = ProductChannelListing.objects.first()
Number of queries 1

In [23]: c.available_for_purchase

In [24]: c.pk
Out[24]: 1

In [25]: ProductChannelListing.objects.filter(available_for_purchase__lt=today).count()
Out[25]: 62
Number of queries 1

In [26]: ProductChannelListing.objects.filter(available_for_purchase__lt=today, pk=1).count()
Out[26]: 0
Number of queries 1
```